roles/activemq/tasks/main.yml

```diff
@@ -1,138 +1,114 @@
+      block:
+        - name: Copy data & config files to Unix FHS dirs
+          copy:
+            remote_src: true
+            src: "{{ item.src }}"
+            dest: "{{ item.dest }}"
+            owner: "{{ username }}"
+            group: "{{ group_name }}"
+            mode: 0755
+          loop:
+            - src: "{{ activemq_home }}/conf"
+              dest: "{{ config_folder }}/activemq"
+            - src: "{{ activemq_home }}/data"
+              dest: "{{ data_folder }}/activemq"
+      rescue:
+        - name: Check Unix FHS dirs
+          stat:
+            path: "{{ item }}"
+          register: amq_dest_dirs
+          loop:
+            - "{{ config_folder }}/activemq"
+            - "{{ data_folder }}/activemq"
+        - name: Handle missing sources
+          ansible.builtin.debug:
+            msg: Looks like the FHS directory structure is already in place
+          when: not amq_dest_dirs.results | map(attribute='stat.exists') | difference([True])
+
+    - name: Remove data & configuration folder from binaries folder
       file:
-        path: "{{ activemq_home }}/data"
+        path: "{{ activemq_home }}/{{ item }}"
         state: absent
+      loop:
+        - conf
+        - data
 
-- name: Allow external connections to webconsole
-  become: true
-  replace:
-    path: "{{ activemq_conf }}/jetty.xml"
-    regexp: "127.0.0.1"
-    replace: "0.0.0.0"
-    owner: "{{ username }}"
-    group: "{{ group_name }}"
+    - name: Allow external connections to webconsole
+      replace:
+        path: "{{ activemq_conf }}/jetty.xml"
+        regexp: "127.0.0.1"
+        replace: "0.0.0.0"
+        owner: "{{ username }}"
+        group: "{{ group_name }}"
 
-- name: Change location of ActiveMQ logs
-  become: true
-  replace:
-    path: "{{ activemq_conf }}/log4j.properties"
-    regexp: "activemq.data"
-    replace: "activemq.log"
-    owner: "{{ username }}"
-    group: "{{ group_name }}"
+    - name: Change location of ActiveMQ logs
+      replace:
+        path: "{{ activemq_conf }}/log4j.properties"
+        regexp: "activemq.data"
+        replace: "activemq.log"
+        owner: "{{ username }}"
+        group: "{{ group_name }}"
 
-- name: Add activemq_home and host to setenv script
-  become: true
-  blockinfile:
-    path: "{{ config_folder }}/setenv.sh"
-    marker: "# {mark} ACTIVEMQ ENV VARS"
-    owner: "{{ username }}"
-    group: "{{ group_name }}"
-    block: |
-      export ACTIVEMQ_HOME={{ activemq_home }}
-      export ACTIVEMQ_HOST={{ activemq_host }}
+    - name: Add activemq_home and host to setenv script
+      blockinfile:
+        path: "{{ config_folder }}/setenv.sh"
+        marker: "# {mark} ACTIVEMQ ENV VARS"
+        owner: "{{ username }}"
+        group: "{{ group_name }}"
+        block: |
+          export ACTIVEMQ_HOME={{ activemq_home }}
+          export ACTIVEMQ_HOST={{ activemq_host }}
+      tags:
+        - molecule-idempotence-notest
```

Comment:
blockinfile isn't idempotent?

Reply:
this isolated task and the  module itself are indeed idempotent. But the file is also handled in the java role using a `template` task. I think that's was created the idempotency issue.
However given I later on made this other taks skip idempotency checks I guess it's OK to revert this one to normal.
Generally speaking, we could agree that in such scenarios we should:
1. tag the root tasks only 
2. migrate to a model using a set of `template` + `assemble` tasks

I'm open to idea.
I think that basically links to OPSEXP-1280 imho